docs/decisions/app/0007-i18n-type-safety.md

```diff
@@ -0,0 +1,36 @@
+# Enable i18next type checks
+
+- Status: proposed
+- Deciders: [list everyone involved in the decision] <!-- optional -->
+- Date: 2023-08-28
+
+## Context and Problem Statement
+
+The codebase utilizes I18next as its internationalization library. [I18next provides the option to enable type checking](https://www.i18next.com/overview/typescript), to help catch references to invalid i18n key paths. This is a feature that projects at Nava have desired before, but the path to implement the feature isn't always the clearest to teams.
+
+## Considered Options
+
+- Enable type checking of I18next
+- Do not enable type checking of I18next
+
+## Decision Outcome
+
+Chosen option: Enable type checking of I18next.
+
+### Positive Consequences
+
+- Type checking will help catch references to invalid i18n key paths, which will help prevent teams from shipping experiences with missing content. Migration efforts that affect locale files will be less risky as a result.
+- Developer tooling will be able to provide better autocomplete suggestions for i18n key paths.
+
+### Negative Consequences
+
+- A TypeScript file representing the i18n namespaces must be maintained alongside the JSON locale files. This file will need to be updated whenever a new locale file is added or removed. If the file isn't updated, a developer will receive type errors when they reference a valid i18n key path. This could be a source of confusion for developers who aren't familiar with the i18next type checking feature.
+  - This file [can be generated](https://github.com/i18next/i18next-resources-for-ts).
+  - Tests can be added to catch missing namespaces in the file.
+- TypeScript errors, particularly those related to i18n, can be difficult to understand. This could slow down developers.
+- Compilation time may be impacted depending on a project's size
+- Type checking only works when the locale files are collocated with the code. If the locale files are stored in a separate repo or hosted by a translation service, the type checking feature will need disabled by removing the `i18next.d.ts` file.
```

Comment:
> i wonder if it would make sense to mention something about handling content that's managed outside the codebase e.g. an external translation service. even if we don't architect for that immediately, could still be helpful to link to a reference like this one https://dev.to/adrai/supercharge-your-typescript-app-mastering-i18next-for-type-safe-translations-2idp that explains how that might be done if a project ends up needing to do that before it's supported out of the box in the template

@lorenyu I tried touching on this here (and included that link below). Is there different phrasing that'd make this clearer?

Reply:
Oh I see, I guess I didn't interpret that article as saying that we'd need to disable typechecking, so maybe say something like "if you have your files stores outside of the repo e.g. in a translation service, then you can either (a) disable type checking by removing the type checking feature, or (b) follow the pattern described in <this article> explaining how to generate an interface file that can be type checked against"

Reply:
Sounds good, updated in https://github.com/navapbc/template-application-nextjs/pull/204/commits/fd55532d12b09f530790d85abea1df484ad0bee8